backend/service-question/src/main/java/g55/cs3219/backend/questionservice/dto/QuestionDto.java

```diff
@@ -0,0 +1,37 @@
+package g55.cs3219.backend.questionservice.dto;
+
+import jakarta.validation.constraints.NotNull;
+import lombok.Getter;
+import lombok.Setter;
+
+import java.io.Serializable;
+import java.util.HashMap;
+import java.util.List;
+
+@Getter
+@Setter
+public class QuestionDto implements Serializable {
+
+    private Integer id;
+
+    @NotNull
+    private String title;
+
+    @NotNull
+    private String description;
+
+    @NotNull
+    private String difficulty;
+
+    @NotNull
+    private List<String> categories;
+
+    @NotNull
+    private List<HashMap<String, String>> examples;
```

Comment:
Actually quite curious why are the examples a List of Hashmap? Do all examples conform to a stricter schema of { input: string, output: string }?

Reply:
Currently, the examples are stored in database with the following format:

```json
"examples": [
    {
        // example 1
        "input": "{input}",
        "output": "{output}",
        "explanation": "{output}"
    },
    {
        // example 2
        "input": "{input}",
        "output": "{output}",
        "explanation": "{output}"
    }
]
```

So most if not all examples conform to this schema. But do let me know if there is any outliers to this case!